lib/request.ts

```diff
@@ -119,89 +130,114 @@ export const request = (
         originalReject(errorWithTimings);
       };
     }
-    const requestObject = httpRequestFn(
-      options,
-      (response: IncomingMessage) => {
+
+    const requestObject = httpRequestFn(options);
+
+    function setReadTimeout() {
+      requestObject.setTimeout(readTimeout, () => {
+        requestObject.socket.destroy();
+        reject(
+          new Error(
+            `Failed to read data within ${readTimeout} ms from ${options.host}`
+          )
+        );
+      });
+    }
+    // Fires once the socket is assigned to a request
+    requestObject.once("socket", (socket: Socket) => {
+      if (options.timing) {
+        timings.socket = getInterval(startTime);
+      }
+      if (socket.connecting) {
+        socket.setTimeout(connectionTimeout, () => {
+          // socket should be manually cleaned up - can use .end() as well
```

Comment:
I don't understand the second part of the comment. What is `can use .end() as well` trying to tell me? Should this code say `socket.end()` in a weird, parallel universe?

Reply:
No, i was trying to mean we can use both socket.destroy and socket.end

I will remove this comment.